File: 프로그래머스/[PCCP 기출문제]/1번_동영상 재생기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <iomanip>

using namespace std;

int timeToSecond(string time) {
    int minutes = stoi(time.substr(0, 2));
    int seconds = stoi(time.substr(3, 2));
    return minutes * 60 + seconds;
}

string secondToTime(int seconds) {
    int minute = seconds / 60;
    int second = seconds % 60;
    stringstream ss;
    ss << setfill('0') << setw(2) << minute << ":" << setfill('0') << setw(2) << second;
    return ss.str();
}

string solution(string video_len, string pos, string op_start, string op_end, vector<string> commands) {

    int videoSecond = timeToSecond(video_len);
    int currentSecond = timeToSecond(pos);
    int opStartSecond = timeToSecond(op_start);
    int opEndSecond = timeToSecond(op_end);

    if (opStartSecond <= currentSecond && currentSecond <= opEndSecond) {
        currentSecond = opEndSecond;
    }

    for(auto command : commands) {
        if (command == "prev") {
            currentSecond = max(0, currentSecond - 10);
        } else if (command == "next") {
            currentSecond = min(videoSecond, currentSecond + 10);
        }
        if (opStartSecond <= currentSecond && currentSecond <= opEndSecond) {
            currentSecond = opEndSecond;
        }
    }

    return secondToTime(currentSecond);
}

File: 프로그래머스/[PCCP 기출문제]/1번_동영상 재생기.java
class Solution {
    int timeToSecond(String time) {
        int minutes = Integer.parseInt(time.substring(0, 2));
        int seconds = Integer.parseInt(time.substring(3, 5));
        return minutes * 60 + seconds;
    }

    String secondToTime(int seconds) {
        int minute = seconds / 60;
        int second = seconds % 60;
        return String.format("%02d:%02d", minute, second);
    }

    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        int videoSecond = timeToSecond(video_len);
        int currentSecond = timeToSecond(pos);
        int opStartSecond = timeToSecond(op_start);
        int opEndSecond = timeToSecond(op_end);

        if (opStartSecond <= currentSecond && currentSecond <= opEndSecond) {
            currentSecond = opEndSecond;
        }

        for(String command : commands) {
            if (command.equals("prev")) {
                currentSecond = Math.max(0, currentSecond - 10);
            } else if (command.equals("next")) {
                currentSecond = Math.min(videoSecond, currentSecond + 10);
            }
            if (opStartSecond <= currentSecond && currentSecond <= opEndSecond) {
                currentSecond = opEndSecond;
            }
        }

        return secondToTime(currentSecond);
    }
}